src/mc_cell_group.hpp

```diff
@@ -87,35 +80,31 @@ class mc_cell_group: public cell_group {
             b.reset();
         }
         lowered_.reset();
-        for (auto& lanes: event_lanes_) {
-            for (auto& lane: lanes) {
-                lane.clear();
-            }
-        }
     }
 
     void set_binning_policy(binning_kind policy, time_type bin_interval) override {
         binners_.clear();
         binners_.resize(gids_.size(), event_binner(policy, bin_interval));
     }
 
-
-    void advance(epoch ep, time_type dt) override {
+    void advance(epoch ep, time_type dt, const event_lane_subrange& event_lanes) override {
         PE("advance");
         EXPECTS(lowered_.state_synchronized());
         time_type tstart = lowered_.min_time();
 
         PE("event-setup");
-        const auto& lc = event_lanes(ep.id);
         staged_events_.clear();
-        for (auto lid: util::make_span(0, gids_.size())) {
-            auto& lane = lc[lid];
-            for (auto e: lane) {
-                if (e.time>=ep.tfinal) break;
-                e.time = binners_[lid].bin(e.time, tstart);
-                auto h = target_handles_[target_handle_divisions_[lid]+e.target.index];
-                auto ev = deliverable_event(e.time, h, e.weight);
-                staged_events_.push_back(ev);
+        // skip event binning if empty lanes are passed
+        if (event_lanes.size()) {
+            for (auto lid: util::make_span(0, gids_.size())) {
+                auto& lane = event_lanes[lid];
+                for (auto e: lane) {
+                    if (e.time>=ep.tfinal) break;
+                    e.time = binners_[lid].bin(e.time, tstart);
```

Comment:
At one point we were talking about lifting the event binning out of the cell groups too. Is this still a good idea? In scope?

Reply:
Probably a good idea. Would best be done as a seperate PR to keep the scope of this PR well-defined, and because moving binning into model would not just be a simple refactoring.